src/Tribe/Editor/Hooks.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Tribe\Events\Editor;
+
+/**
+ * Events block editor hooks.
+ *
+ * @since TBD
+ */
+class Hooks {
+	/**
+	 * Binds and sets up implementations.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
+		$this->add_actions();
+	}
+
+	/**
+	 * Adds the actions required by each Views v2 component.
+	 *
+	 * @since TBD
+	 */
+	protected function add_actions() {
+		add_action( 'current_screen', [ $this, 'add_widget_resources' ] );
+	}
+
+	/**
+	 * Adds the editor styles required by the block editor.
+	 *
+	 * We trigger this action so that we only need to do the is_block_editor() check once.
+	 *
+	 * @since TBD
+	 */
+	public function add_widget_resources() {
+		if ( ! get_current_screen()->is_block_editor() ) {
+			return;
+		}
+
+		add_action( 'admin_print_styles', [ $this, 'admin_print_widget_styles' ] );
+		add_action( 'admin_print_scripts', [ $this, 'admin_print_widget_scripts' ] );
+		add_action( 'admin_print_footer_scripts', [ $this, 'admin_print_footer_widget_scripts' ] );
+		add_action( 'admin_footer', [ $this, 'admin_footer_widgets' ] );
+	}
```

Comment:
Not necessarily a change required here, just wanted to know why all of this is needed. Seems like it could trigger fatals, from third-party triggering actions here that expect to be on the actual Widgets Page.

Reply:
If I'm understanding correctly, these all ensure that the widget styles and JS that are present on the widgets admin page are _also_ present on the block editor so that widget interactions behave normally. The subsequent `do_action()` calls bring in the necessary markup.

A potential option for later is to explicitly enqueue the things we need.